pkg/controller/servicebinding/secret.go

```diff
@@ -48,35 +86,41 @@ func (s *secret) createOrUpdate(payload map[string][]byte, ownerReference metav1
 
 	resourceClient := s.buildResourceClient()
 
-	logger.Debug("Attempt to create secret...")
 	existingSecret, err := s.get()
 	if err != nil {
 		if errors.IsNotFound(err) {
-			_, err := resourceClient.Create(u, metav1.CreateOptions{})
+			secretNew, err := resourceClient.Create(u, metav1.CreateOptions{})
 			if err != nil {
 				logger.Error(err, "Error creating secret")
 				return nil, err
 			}
 			logger.Info("Secret created")
-			return u, nil
+			return secretNew, nil
 		}
 		return nil, err
 	}
+
 	existingSecretData, _, _ := unstructured.NestedMap(existingSecret.Object, "data")
-	payloadInterim := make(map[string]interface{})
-	for k, v := range payload {
-		payloadInterim[k] = base64.StdEncoding.EncodeToString(v)
+
+	existingData := make(map[string][]byte)
+	for k, v := range existingSecretData {
+		value := []byte(fmt.Sprintf("%v", v.(interface{})))
+		existingData[k] = value
 	}
-	comparisonResult := nestedMapComparison(existingSecretData, payloadInterim)
-	if comparisonResult.Success {
-		logger.Debug("Secret data is same. Skip Update")
-	} else {
-		logger.Info("Secret data is different; update secret", "Diff", comparisonResult.Diff)
-		_, err = resourceClient.Update(u, metav1.UpdateOptions{})
+
+	existingSecretHash, err := name(existingSecret.GetName(), existingData)
```

Comment:
Don't bother changing this now, but let's think harder about names please. I understand the original name might not have been everyone's cup of tea, but having a function called `name` is really suboptimal and error prone.

Reply:
@pedjak @isutton what about `hashedName` or `uniqueName`?